apps/prairielearn/src/server.js

```diff
@@ -1319,8 +1319,20 @@ module.exports.initExpress = function () {
   //////////////////////////////////////////////////////////////////////
   // Student pages /////////////////////////////////////////////////////
 
-  // Exam/Homeworks student routes are polymorphic - they have multiple handlers, each of
-  // which checks the assessment type and calls next() if it's not the right type
+  if (isEnterprise()) {
+    app.use('/pl/course_instance/:course_instance_id/upgrade', [
+      require('./ee/pages/studentCourseInstanceUpgrade/studentCourseInstanceUpgrade').default,
+    ]);
```

Comment:
In the future, this could become the "enroll" page discussed here: https://github.com/PrairieLearn/PrairieLearn/pull/8214#discussion_r1272879110